### PR TITLE
fix(findingID): remove duplicate finding IDs

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Remove duplicate finding IDs by making sure the resource ID is unique in the findings.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jfagoagas — DISMISSED** [review]
This looks reasonable to me, thanks @sergargar !

Please review the tests since some are failing.

File: tests/providers/aws/services/iam/iam_disable_45_days_credentials/iam_disable_45_days_credentials_test.py [hunk 1]
```diff
@@ -380,17 +380,17 @@ def test_user_both_access_keys_not_used(self):
...
                     == f"User {user} has not used access key 1 in the last 45 days (100 days)."
                 )
-                assert result[1].resource_id == user
-                assert result[1].resource_arn == arn
-                assert result[1].region == AWS_REGION
-                assert result[2].status == "FAIL"
+                assert result[-1].resource_id == user + "/AccessKey2"
+                assert result[-1].resource_arn == arn
+                assert result[-1].region == AWS_REGION
+                assert result[-2].status == "FAIL"
```
Comment (MrCloudSec) [inline tests/providers/aws/services/iam/iam_disable_45_days_credentials/iam_disable_45_days_credentials_test.py]:
```suggestion
                assert result[1].resource_id == user + "/AccessKey2"
                assert result[1].resource_arn == arn
                assert result[1].region == AWS_REGION
                assert result[2].status == "FAIL"
```

File: tests/providers/aws/services/iam/iam_disable_45_days_credentials/iam_disable_45_days_credentials_test.py [hunk 2]
```diff
@@ -380,17 +380,17 @@ def test_user_both_access_keys_not_used(self):
...
                 assert (
                     result[2].status_extended
                     == f"User {user} has not used access key 2 in the last 45 days (100 days)."
                 )
-                assert result[2].resource_id == user
-                assert result[2].resource_arn == arn
-                assert result[2].region == AWS_REGION
+                assert result[-2].resource_id == user + "/AccessKey1"
+                assert result[-2].resource_arn == arn
+                assert result[-2].region == AWS_REGION
```
Comment (MrCloudSec) [inline tests/providers/aws/services/iam/iam_disable_45_days_credentials/iam_disable_45_days_credentials_test.py]:
```suggestion
                assert result[2].resource_id == user + "/AccessKey1"
                assert result[2].resource_arn == arn
                assert result[2].region == AWS_REGION
```

File: tests/providers/aws/services/iam/iam_disable_30_days_credentials/iam_disable_30_days_credentials_test.py [hunk 3]
```diff
@@ -380,18 +380,18 @@ def test_user_both_access_keys_not_used(self):
                     result[1].status_extended
                     == f"User {user} has not used access key 1 in the last 30 days (100 days)."
                 )
-                assert result[1].resource_id == user
-                assert result[1].resource_arn == arn
-                assert result[1].region == AWS_REGION
+                assert result[-1].resource_id == user + "/AccessKey2"
+                assert result[-1].resource_arn == arn
+                assert result[-1].region == AWS_REGION
```
Comment (MrCloudSec) [inline tests/providers/aws/services/iam/iam_disable_30_days_credentials/iam_disable_30_days_credentials_test.py]:
```suggestion
                assert result[1].resource_id == user + "/AccessKey2"
                assert result[1].resource_arn == arn
                assert result[1].region == AWS_REGION
```

File: tests/providers/aws/services/iam/iam_disable_30_days_credentials/iam_disable_30_days_credentials_test.py [hunk 4]
```diff
@@ -380,18 +380,18 @@ def test_user_both_access_keys_not_used(self):
...
                 assert (
                     result[2].status_extended
                     == f"User {user} has not used access key 2 in the last 30 days (100 days)."
                 )
-                assert result[2].resource_id == user
-                assert result[2].resource_arn == arn
-                assert result[2].region == AWS_REGION
+                assert result[-2].resource_id == user + "/AccessKey1"
+                assert result[-2].resource_arn == arn
+                assert result[-2].region == AWS_REGION
```
Comment (MrCloudSec) [inline tests/providers/aws/services/iam/iam_disable_30_days_credentials/iam_disable_30_days_credentials_test.py]:
```suggestion
                assert result[2].resource_id == user + "/AccessKey1"
                assert result[2].resource_arn == arn
                assert result[2].region == AWS_REGION
```

File: tests/providers/aws/services/iam/iam_disable_30_days_credentials/iam_disable_30_days_credentials_test.py [hunk 5]
```diff
@@ -325,9 +325,9 @@ def test_user_access_key_2_not_used(self):
                     result[1].status_extended
                     == f"User {user} has not used access key 2 in the last 30 days (100 days)."
                 )
-                assert result[1].resource_id == user
-                assert result[1].resource_arn == arn
-                assert result[1].region == AWS_REGION
+                assert result[-1].resource_id == user + "/AccessKey2"
+                assert result[-1].resource_arn == arn
+                assert result[-1].region == AWS_REGION
```
Comment (MrCloudSec) [inline tests/providers/aws/services/iam/iam_disable_30_days_credentials/iam_disable_30_days_credentials_test.py]:
```suggestion
                assert result[1].resource_id == user + "/AccessKey2"
                assert result[1].resource_arn == arn
                assert result[1].region == AWS_REGION
```

File: tests/providers/aws/services/iam/iam_disable_30_days_credentials/iam_disable_30_days_credentials_test.py [hunk 6]
```diff
@@ -275,9 +275,9 @@ def test_user_access_key_1_not_used(self):
                     result[1].status_extended
                     == f"User {user} has not used access key 1 in the last 30 days (100 days)."
                 )
-                assert result[1].resource_id == user
-                assert result[1].resource_arn == arn
-                assert result[1].region == AWS_REGION
+                assert result[-1].resource_id == user + "/AccessKey1"
+                assert result[-1].resource_arn == arn
+                assert result[-1].region == AWS_REGION
```
Comment (MrCloudSec) [inline tests/providers/aws/services/iam/iam_disable_30_days_credentials/iam_disable_30_days_credentials_test.py]:
```suggestion
                assert result[1].resource_id == user + "/AccessKey1"
                assert result[1].resource_arn == arn
                assert result[1].region == AWS_REGION
```

**Review by jfagoagas — DISMISSED** [review]
👍

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**